crates/rpc/src/types/errors.rs

```diff
@@ -23,10 +23,24 @@ pub enum ApiError {
     ValidatorNotFound(String),
 
     #[error("Too many validator IDs in request")]
-    TooManyValidatorsIds(),
+    TooManyValidatorsIds,
 }
 
-impl Reject for ApiError {}
+// impl ResponseError trait allows to convert our errors into http responses with appropriate data
+impl ResponseError for ApiError {
+    fn error_response(&self) -> HttpResponse {
+        match self {
+            Self::TooManyValidatorsIds => {
+                HttpResponse::InternalServerError().json("Too many validator IDs in request")
+            }
+            _ => {
+                todo!()
+            }
+        }
+    }
+}
+
+// impl Reject for ApiError {}
```

Comment:
You removed all of our usages of APIError, are we going to use our type, if not we should probably remove ``ApiError`

Reply:
Ideally we would use our ApiError Enum like we did before then, the code would map our error type to actix like we did for warp

Reply:
In the documentation here it shows how we can define our own errors https://actix.rs/docs/errors#an-example-of-a-custom-error-response